src/HttpReply.cc

```diff
@@ -304,7 +304,8 @@ HttpReply::hdrCacheInit()
     date = header.getTime(Http::HdrType::DATE);
     last_modified = header.getTime(Http::HdrType::LAST_MODIFIED);
     surrogate_control = header.getSc();
-    content_range = header.getContRange();
+    if (sline.status() == Http::scPartialContent)
+        content_range = header.getContRange();
```

Comment:
That would mean that content_range can be non-null here, and both changed/unchanged code is wrong because it leaks memory. So, either we should 'assert(!content_range)' there before initialization(unlikely), or delete non-null content_range. The same is true for other fields, consuming heap memory, such as cache_control and surrogate_control, so I would call HttpReply::hdrCacheClean() at the beginning of HttpReply::hdrCacheInit() and remove corresponding ternary operators.

Reply:
Let's just keep the method code consistent for now. I doubt this code is buggy (i.e., called for a dirty object), and investigating/fixing those alleged bugs is outside this project scope.

Reply:
Ok, however, note that there is a place where hdrCacheInit() is called for a dirty object: HttpReply::updateOnNotModified(). Currently, this code is correct since that method does hdrCacheClean() itself.

Reply:
A hdrCacheClean() call ensures that a subsequent hdrCacheInit() call operates on a clean object (clean from these two methods point of view, which is naturally limited to "cached" values like `content_range`). In other words, calling hdrCacheClean() first avoids the "dirty" object state that a subsequent hdrCacheInit() cannot handle.